src/DataType.cpp

```diff
@@ -42,6 +44,35 @@ std::string data_type_to_string(DataType dtype) {
     return str;
 }
 
+DataType string_to_data_type(std::string dtype) {
+
+    struct data_type_map : public std::map<std::string, DataType> {
+        data_type_map() {
+            (*this)["bool"] = DataType::Bool;
+            (*this)["char"] = DataType::Char;
+            (*this)["float"] = DataType::Float;
+            (*this)["single"] = DataType::Float;
+            (*this)["double"] = DataType::Double;
+            (*this)["int8"] = DataType::Int8;
+            (*this)["int16"] = DataType::Int16;
+            (*this)["int32"] = DataType::Int32;
+            (*this)["int64"] = DataType::Int64;
+            (*this)["uint8"] = DataType::UInt8;
+            (*this)["uint16"] = DataType::UInt16;
+            (*this)["uint32"] = DataType::UInt32;
+            (*this)["uint64"] = DataType::UInt64;
+            (*this)["string"] = DataType::String;
+            (*this)["nothing"] = DataType::Nothing;
+        }
+    };
```

Comment:
Thanks for the comment. So here the idea was to make it user/system convenient and accept all cases like "Double", "double", "DOUBLE" whatever "douBLe" etc. Such that, for instance, the Matlab "double" doesn't need to be converted to "Double" as it's clear it's a double type. Otherwise one has to go always do to the NIX code and see how exactly the string representation should be specified. What do you think?